src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDataReader.cs

```diff
@@ -2582,55 +2582,104 @@ private T GetFieldValueInternal<T>(int i)
 
         private T GetFieldValueFromSqlBufferInternal<T>(SqlBuffer data, _SqlMetaData metaData)
         {
-            Type typeofT = typeof(T);
-            if (_typeofINullable.IsAssignableFrom(typeofT))
+            // this block of type specific shortcuts uses RyuJIT jit behaviours to achieve fast implementations of the primative types
```

Comment:
This comment explains why the type specific paths are setup the way they are.

Reply:
nit: spelling, primitive (here and elsewhere)

Reply:
Fixed, thanks.